### PR TITLE
Run `cargo clippy --fix`

### PR DESCRIPTION
**Pull request by theHamsta**
Using variables names directly in format strings might increase the minimum required Rust version and could be excluded from clippy lints.

Please let me know whether this is in scope for this repo. Clippy seems to do quite some readability improvements.

### CODE REVIEW

**Comment by Lyude** [comment]
Hey! Sorry, I've still got to take a closer look at your gio contribution as well.

And yes - clippy would absolutely be in scope, I had been thinking of starting work on this and making it so we actually start running clippy in our CI. I was thinking I might have to do some work to figure out which nags we did and didn't want, but it seems like clippy has some rather sensible defaults so it doesn't seem like that'll be needed :).

Anyway - I took a look through all of the changes it made and honestly these seem fine to me, if you can get it passing the rust-fmt check I'm happy to merge it

**Comment by Lyude** [comment]
Hey, sorry that it kept giving you issues with letting you run the CI, I think that should be fixed now so it should let you run it whenever you want

**Review by Lyude — CHANGES_REQUESTED** [review]
Also would you mind squashing this into https://github.com/Lyude/neovim-gtk/pull/67/commits/af1d38f77aae3b3cac922fec61452d2006ba22f2 ?

**Review by Lyude — APPROVED** [review]

**Comment by theHamsta** [comment]
`export CARGO_UNSTABLE_SPARSE_REGISTRY=true` would probably make the checkout of the cargo registry faster.

@Lyude I saw that there are more lints than the ones that can be fixed by `cargo clippy --fix`. Please check the manual fixes that I applied to  fix 

wrong_self_convention
unnecessary_mut_passed
borrowed_box
needless_lifetimes
match_like_matches_macro

I allowed lints that are probably larger refactorings and should be tackled on a one-by-one basis.

File: src/main.rs [hunk 1]
```diff
@@ -1,5 +1,11 @@
 #![windows_subsystem = "windows"]
 
+#![allow(clippy::new_without_default)]
+#![allow(clippy::type_complexity)]
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::comparison_chain)]
+#![allow(clippy::await_holding_refcell_ref)]
```
Comment (theHamsta) [inline src/main.rs:7]: those probably, require refactors. `new_without_default` could be fixed by a few `derive(Default)` if desired.

Comment (Lyude) [reply src/main.rs:6]: This lgtm, thanks!

**Pull request merged**